### ISSUE TITLE
twisted.names.client.lookupAuthority displays authoritative name servers as 'auth=False'.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5421]
|<img alt="uberj's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| uberj reported|
|-|-|
|Trac ID|trac#5421|
|Type|defect|
|Created|2011-12-06 06:03:48Z|
|Branch|https://github.com/twisted/twisted/tree/rrheader-auth-5421|

If you want to test this out, here is a script.

```python
#!/usr/bin/env python

from twisted.names import client
from twisted.internet import reactor
import sys

r = client.Resolver(servers=[('128.193.0.10',53)])

def test_soa(a):
    print a
    reactor.stop()


if __name__ == '__main__':
    r.lookupAuthority(sys.argv[1]).addCallback(test_soa)
    reactor.run()
```

Here is the output along with a dig showing the correct results.
```
uberj@uberj ~/Downloads/Twisted-11.1.0 (> ^ ^)> python soa_bug.py oregonstate.edu
([<RR name=oregonstate.edu type=SOA class=IN ttl=86400s auth=False>], [<RR name=oregonstate.edu type=NS class=IN ttl=86400s auth=False>, <RR name=oregonstate.edu type=NS class=IN ttl=86400s auth=False>, <RR name=oregonstate.edu type=NS class=IN ttl=86400s auth=False>, <RR name=oregonstate.edu type=NS class=IN ttl=86400s auth=False>, <RR name=oregonstate.edu type=NS class=IN ttl=86400s auth=False>], [<RR name=dns.merit.net type=A class=IN ttl=4294s auth=False>, <RR name=ns1.nero.net type=A class=IN ttl=4289s auth=False>, <RR name=ns1.ucsb.edu type=A class=IN ttl=1797s auth=False>, <RR name=ns1.ucsb.edu type=AAAA class=IN ttl=961s auth=False>, <RR name=ns1.oregonstate.edu type=A class=IN ttl=86400s auth=False>, <RR name=ns2.oregonstate.edu type=A class=IN ttl=86400s auth=False>])

uberj@uberj ~/Downloads/Twisted-11.1.0 (> ^ ^)> dig +short @ns2.oregonstate.edu oregonstate.edu soa
ns1.oregonstate.edu. hostmaster.oregonstate.edu. 1323150914 300 900 604800 86400

```

Here is a patch. Credit goes to exarkun in #twisted.
```
#!diff
Index: twisted/names/dns.py
===================================================================
--- twisted/names/dns.py        (revision 33284)
+++ twisted/names/dns.py        (working copy)
@@ -1635,7 +1635,7 @@

     def parseRecords(self, list, num, strio):
         for i in range(num):
-            header = RRHeader()
+            header = RRHeader(auth=self.auth)
             try:
                 header.decode(strio)
             except EOFError:

```

<details><summary>Searchable metadata</summary>

```
trac-id__5421 5421
type__defect defect
reporter__uberj uberj
priority__normal normal
milestone__ 
branch__branches_rrheader_auth_5421 branches/rrheader-auth-5421
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__names names
keywords__lookupAuthority lookupAuthority
time__1323151428000000 1323151428000000
changetime__1325946337000000 1325946337000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [33381]) Branching to 'rrheader-auth-5421'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/rrheader-auth-5421)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set owner to @exarkun|
|-|-|

Looks great.  A very nice, straightforward fix.  Please land.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [33417]) Merge rrheader-auth-5421 

Author: exarkun
Reviewer: glyph
Fixes: [#5421](https://github.com/twisted/twisted/issues/5421)

When constructing `RRHeader` instances while parsing a message into a
`twisted.names.dns.Message` instance, set the `auth` flag on the `RRHeader`
instead to match the auth flag on the `Message` instance.

This makes `RRHeader.auth` reflect whether the record is part of an
authoritative message or not.
